### ISSUE TITLE
Alternatives for iOS7

### ISSUE THREAD

**Issue by georg90** [plamoni/SiriProxy#572]
Since the other topic gets flooded, we could continue discussing options for iOS 7 here!

For now (as I see it) we only have the google/yahoo option: so ask Siri to Google/yahoo something and redirect this request to your own web server to execute commands.

**Comment by X-Mort-X** [comment]
Hey, just curious about the other viable options we have?
Not a fan about the implementation of the 'google/yahoo' option seems messy to me, of course its a clever option but just personally not a viable option. 
So that being said, are there any other options available that are not so much of a work around and more of an implementation? 
cheers guys!

**Comment by Timvdv** [comment]
Well, another option is jailbreaking you iDevice. I don't have a jailbroken device so I cannot confirm if this will work. But I think it should be possible to make Siri think it's on IOS 6 instead of IOS 7. But remember, jailbreaking is dangerous from a security perspective.

**Comment by X-Mort-X** [comment]
Nah not willing to do that lol, I was considering buying a 4s on ios 6 but id prefer not too lol
what about other voice applications that run through Linux or something like that? although I know that's not siri lol but is there not an alternative that's already open source that's as good as siri?

**Comment by Timvdv** [comment]
I recently came across this project: http://jasperproject.github.io
It looks pretty decent, but I don't like the voice. I'd like to know how the voice recognition is.

**Comment by X-Mort-X** [comment]
ah it looks good, your right though, that voice is awefull, wonder if you can change it.....
cheers for the info!

**Comment by yanniks** [comment]
I started coding a simple one (I'm going to extend it in the near future) about 1.5 weeks ago, it's web based: https://github.com/yanniks/everassist

For dictation, use the small button with the mic on your keyboard.

**Comment by yanniks** [comment]
And currently, it's German only, but I'll extend that.

**Comment by jumper4000** [comment]
can these alternatives work with Bing instead of Yahoo/Google? At least with Bing, you don't have to unlock your phone and Siri doesn't need to launch a browser.

**Comment by X-Mort-X** [comment]
Just tried using bing and it still needs to unlock the iphone......
@yanniks: I don't understand German lol but let us know when you port it to English and I'll give it a go if you like :)

**Comment by jumper4000** [comment]
@X-Mort-X that's weird. Bing is Siri's default search engine. On mine, when I say "Bing", then any command, it just shows the result right there on lock screen. Try saying something like, "Bing where can I buy a TV from?"

**Comment by georg90** [comment]
I tried to use bing as well, problem here is that the iPhone will not call a website to get the results, it simply does some magic inside (used wire shark) and connect to bing once, and the rest of polls is to apple servers...
So they built it inside of iOS...
Same when you ask for "bing some command"

**Comment by X-Mort-X** [comment]
Mine does do that for some queries but not others, seems to be for some keywords it will call up a website via bing and others it's displayed in the lock screen, just depends on what I say!

**Comment by lembi0** [comment]
mine does not :).
Neither with Google nor with Bing... hm (iOS6)

**Comment by Timvdv** [comment]
Hmm, for me (IOS7) everything happens within the 'Siri' window when I search with Bing

**Comment by lembi0** [comment]
iOS6 opens Bing in a browser window (like Google)

**Comment by lembi0** [comment]
Best - or "closest to Siri" solution - i think - would be a voice recognizing App which is able to send out HTTP requests (open an URL with the spoken text as querry-string)  to a self definable URL (raspberry), parse the response an wirte/read it.

**Comment by georg90** [comment]
That's exactly what most people don't want, the key feature (my opinion) of siriproxy was the integration with Siri directly, so you can start it from any location etc. and it feels  "build in"
There are already some other apps doing what you describe, you just need to port these web apps to iOS..

**Comment by lembi0** [comment]
Well sure, Siri is a little more usable but i think it's just a "Problem" of how the App works.

In my opinion, the WebApps you mentioned, are not pretty user friendly. You first have to start the App then push a button, push the small mic. button on the keyboard, you don't have a spoken response.... and worst of all: your requests have to start with "Google" ... or something like that.

If there is an App which you just have to start and say something (and get a spoken response)... the main difference is only that you can't use it on the lock screen.... that's all. The rest will be as comfortable as using SiriProxy...?! ...I would use something like this!

...but the proxy won't work until someone gets the original Apple cert, guzzoni is using (my opinion). ...and if you have this cert... it's because someone stole it, which makes the system kind of illegal. I've spent about 150 hours to figure out the problem with the proxy and where it breakes ....in my opinion, there is no "legal" way and no way without a jailbrake, to bring it back to work like it worked before.

**Comment by georg90** [comment]
Second down point: The home button

Afk on a non jailbreaked device you cannot use the home button to trigger the app.

**Comment by georg90** [comment]
Yes the web apps are complicated to use, I just wanted to say that there is already a way around it, even if it isn't as convenient as Siri 
Can you use the microphone with an app without the keyboard and without pressing a button?

**Comment by lembi0** [comment]
No, you can't use the home button... but the "quick start" menu at the lower end of the display (how is this called?)
Were is the difference in pushing a button on the display or pushing the home button? :)  

I don't know anything about Apps, but i think it should be possible to use the mic without pressing a specific button (not in WebApps) ... why not?

**Comment by Timvdv** [comment]
Just to be clear: Is it possible to alter Siriproxy so it will a be intercepting and reading the requests?

This way most of the Siriproxy commands can still be used, only without a response.

**Comment by lembi0** [comment]
No it's not. Siri needs a proper response from guzzoni to work correctly...  without a valid session ID (exchanged between Siri and Guzzoni), Siri will not send out speech requests.

**Comment by X-Mort-X** [comment]
So if it's a problem with the cert.... And that's on every legit iphone running ios 7..... Can't you simply clone it in some manner a send it from the proxy itself?

**Comment by X-Mort-X** [comment]
Oh and the way round the problem of pressing the button for me would be to pull apart the iphone and have a specific word that actuated a electronic switch replacing the home button all together! But obviously that would only happen if I deployed it house wide, I'd also have mic's everywhere lol

**Comment by lembi0** [comment]
the cert is on the server... and if it would be "cloneable" it would be no great saftey feature, i think...  :)

**Comment by X-Mort-X** [comment]
Well I suppose it depends if it's designed as a safety feature lol i don't actually know the workings of siri I was just hoping to spur a moment of genius in someone lol clearly that didn't work haha, surely though, if the cert is only located on the server, then there must be authentication keys on each device? With that in mind can't a key gem be made like they use on cracked software?

**Comment by X-Mort-X** [comment]
Right I have been trying this voice program 'jasper'! Cannot make it work! It won't connect to the wireless at all, can't seem to pick up the ad-hoc network it supposedly broadcasts on my win7 lappy or my iphone! Getting a bit peeved lol

**Comment by bgottsch** [comment]
Don't know if any one saw the WWDC 14 Keynote, but there is a new home automation API with Siri integration. The info still is not much but it may be the solution Apple is providing. Also Siri has a new always-on feature where you say "Hey Siri" and it activates, which is cool to combine with automation and controlling stuff. As I mentioned, there is still a lot to learn with this new programming language and API's, so it might take a while. Also iOS 8 is not yet released so discussing about it is forbidden by Apple.

**Comment by nelmedeiros** [comment]
Is the the end of SiriProxy project!!—

**Comment by hackworth** [comment]
I had need for a generic voice command processor so I've completely gutted SiriProxy and created [SPVoice](https://github.com/Hackworth/SPVoice) from it. It uses SiriProxy plugins with very little modification, since I'd hate to lose all of the work I've done on plugins over the past year or two. 

[SPVoice](https://github.com/Hackworth/SPVoice) can be used as either a Ruby class, for any of you Ruby programmers, and can also be used as an HTTP JSON API. I've created a fork of [Siri-API](https://github.com/Hackworth/Siri-API) configured out of the box to work with [SPVoice](https://github.com/Hackworth/SPVoice) with a few other modifications. Saying "Yahoo watch game of thrones" does exactly that. I've converted an [XBMC Plugin](https://github.com/Hackworth/SPVoice-XBMC)

If you end up using it, feel free to send me some feedback!
